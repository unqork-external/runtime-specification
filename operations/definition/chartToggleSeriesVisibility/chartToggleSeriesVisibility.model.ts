import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { ChartToggleSeriesVisibilityOperationOptions } from './chartToggleSeriesVisibility.options'
import { OpsBuilderStable, Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@Stable()
@OpsBuilderStable()
@Description('Updates series visibility on a Chart component.')
@DiscriminatorValue(OperationTypes.CHART_TOGGLE_SERIES_VISIBILITY)
export class ChartToggleSeriesVisibilityOperation extends Operation<OperationTypes.CHART_TOGGLE_SERIES_VISIBILITY> {
  @Required()
  public type = OperationTypes.CHART_TOGGLE_SERIES_VISIBILITY as const

  @Required()
  public options: ChartToggleSeriesVisibilityOperationOptions = new ChartToggleSeriesVisibilityOperationOptions()
}
