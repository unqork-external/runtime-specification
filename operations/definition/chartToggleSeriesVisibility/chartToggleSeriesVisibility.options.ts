import { CollectionOf, Description, Example, Required } from '@tsed/schema'

export class ChartToggleSeriesVisibilityOperationOptions {
  @Required()
  @Description('The Property ID for the target Chart component.')
  targetKey: string

  @Required()
  @CollectionOf('string')
  @Description('The IDs of the series to target the visibility toggle.')
  @Example(['seriesA', 'seriesB'])
  targetIds: string[]

  @Required()
  @Description('The new state of the series visibility.')
  visible: boolean
}
