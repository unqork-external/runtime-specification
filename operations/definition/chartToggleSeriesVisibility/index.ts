export * from './chartToggleSeriesVisibility.model'
export * from './chartToggleSeriesVisibility.options'
