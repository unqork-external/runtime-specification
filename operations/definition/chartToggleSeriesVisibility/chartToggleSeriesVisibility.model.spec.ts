import { ChartToggleSeriesVisibilityOperation } from './chartToggleSeriesVisibility.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ChartToggleSeriesVisibility Tests', () => {
  it('ChartToggleSeriesVisibility Model', () => {
    const { schema } = generateSchemaAndValidate(ChartToggleSeriesVisibilityOperation)

    expect(schema).toMatchSnapshot()
  })
})
