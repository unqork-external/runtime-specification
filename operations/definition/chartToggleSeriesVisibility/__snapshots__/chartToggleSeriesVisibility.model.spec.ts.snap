// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: ChartToggleSeriesVisibility Tests ChartToggleSeriesVisibility Model 1`] = `
{
  "definitions": {
    "ChartToggleSeriesVisibilityOperationOptions": {
      "properties": {
        "targetIds": {
          "description": "The IDs of the series to target the visibility toggle.",
          "examples": [
            [
              "seriesA",
              "seriesB",
            ],
          ],
          "items": {
            "type": "string",
          },
          "type": "array",
        },
        "targetKey": {
          "description": "The Property ID for the target Chart component.",
          "minLength": 1,
          "type": "string",
        },
        "visible": {
          "description": "The new state of the series visibility.",
          "type": "boolean",
        },
      },
      "required": [
        "targetKey",
        "targetIds",
        "visible",
      ],
      "type": "object",
    },
  },
  "description": "Updates series visibility on a Chart component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ChartToggleSeriesVisibilityOperationOptions",
    },
    "type": {
      "const": "CHART_TOGGLE_SERIES_VISIBILITY",
      "examples": [
        "CHART_TOGGLE_SERIES_VISIBILITY",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
